api/nodes/filters.py

```diff
@@ -13,23 +13,18 @@
 class NodesListFilterMixin(ODMFilterMixin):
 
     def _operation_to_query(self, operation):
-        # We special case filters on root because root isn't a field; to get the children
-        # of a root, we use a custom manager method, Node.objects.get_children, and build
-        # a query from that
         if operation['source_field_name'] == 'root':
-            child_pks = []
-            for root_guid in operation['value']:
-                if not root_guid:
-                    raise InvalidFilterValue(value=root_guid)
-                root = utils.get_object_or_error(Node, root_guid, display_name='root', prefetch_fields=self.serializer_class().model_field_names)
-                child_pks.extend(Node.objects.get_children(root=root, primary_keys=True))
-            return Q('id', 'in', child_pks)
-        elif operation['source_field_name'] == 'parent_node':
-            if operation['value']:
-                parent = utils.get_object_or_error(Node, operation['value'], display_name='parent', prefetch_fields=self.serializer_class().model_field_names)
-                return Q('parent_nodes', 'eq', parent.id)
-            else:
-                return Q('parent_nodes', 'isnull', True)
+            if None in operation['value']:
+                raise InvalidFilterValue()
+            return functools.reduce(operator.or_, [
+                Q('root__guids___id', 'eq', value)
```

Comment:
Ah, hadn't tried this. It does work and I made that change in 915b8ea. 

Also updated the `parent` case to (properly) exclude linked nodes. 

Tests will (maybe, hopefully) pass this time.